db/migrate/20250227161653_populate_comments_with_restricted_visibility_permissions.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require Rails.root.join("db/migrate/migration_utils/permission_adder")
+
+class PopulateCommentsWithRestrictedVisibilityPermissions < ActiveRecord::Migration[7.1]
+  def up
+    # grant the permission to view/write/edit comments with restricted visibility
+    # to users that can manage work package relations
+    ::Migration::MigrationUtils::PermissionAdder.add(:manage_work_package_relations,
+                                                     :view_comments_with_restricted_visibility)
+    ::Migration::MigrationUtils::PermissionAdder.add(:manage_work_package_relations,
+                                                     :write_comments_with_restricted_visibility)
+    ::Migration::MigrationUtils::PermissionAdder.add(:manage_work_package_relations,
+                                                     :edit_own_comments_with_restricted_visibility)
```

Comment:
👍🏾  we seem to be reaching a point where the `.add()` method should accept many permissions for a simpler and implicitly atomic interface.